src/main/java/com/solvd/movie/kafka/consumer/KafkaConsumer.java

```diff
@@ -0,0 +1,46 @@
+package com.solvd.movie.kafka.consumer;
+
+import com.solvd.movie.model.Event;
+import com.solvd.movie.model.exception.IllegalActionException;
+import com.solvd.movie.service.EsMovieService;
+import jakarta.annotation.PostConstruct;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import reactor.kafka.receiver.KafkaReceiver;
+import reactor.kafka.receiver.ReceiverOffset;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class KafkaConsumer {
+
+    private final KafkaReceiver<String, Event> kafkaReceiver;
+    private final EsMovieService esMovieService;
+
+    @PostConstruct
+    public void receive() {
+        this.kafkaReceiver.receive()
+                .subscribe(record -> {
+                    ReceiverOffset offset = record.receiverOffset();
+                    Event event = record.value();
+                    log.info("Received event: " + event.toString());
+                    switch (event.getAction()) {
```

Comment:
@hizmailovich solve problem with gigantic `switch-cases`, which is not elegant solution, by using map of spring beans